aws_lambda_builders/workflows/nodejs_npm/workflow.py

```diff
@@ -40,18 +44,34 @@ def __init__(self, source_dir, artifacts_dir, scratch_dir, manifest_path, runtim
             tar_dest_dir, scratch_dir, manifest_path, osutils=osutils, subprocess_npm=subprocess_npm
         )
 
-        npm_install = NodejsNpmInstallAction(artifacts_dir, subprocess_npm=subprocess_npm)
-
         npm_copy_npmrc = NodejsNpmrcCopyAction(tar_package_dir, source_dir, osutils=osutils)
 
         self.actions = [
             npm_pack,
             npm_copy_npmrc,
             CopySourceAction(tar_package_dir, artifacts_dir, excludes=self.EXCLUDED_FILES),
-            npm_install,
-            NodejsNpmrcCleanUpAction(artifacts_dir, osutils=osutils),
         ]
 
+        if self.download_dependencies:
+            # installed the dependencies into artifact folder
+            self.actions.append(NodejsNpmInstallAction(artifacts_dir, subprocess_npm=subprocess_npm))
+
+            # if dependencies folder exists, copy dependencies into dependencies into dependencies folder
+            if self.dependencies_dir:
+                self.actions.append(CopyDependenciesAction(source_dir, artifacts_dir, self.dependencies_dir))
+        else:
+            # if dependencies folder exists and not download dependencies, simply copy the dependencies from the
+            # dependencies folder to artifact folder
+            if self.dependencies_dir:
+                self.actions.append(CopySourceAction(self.dependencies_dir, artifacts_dir))
+            else:
+                LOG.info(
+                    "download_dependencies is false and the dependencies_dir is not exists, just copying the source "
```

Comment:
nit: "download_dependencies is False and dependencies_dir is None. Copying the source files into the artifacts directory."